Purchases/Networking/Operations/GetOfferingsOperation.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  GetOfferingsOperation.swift
+//
+//  Created by Joshua Liebowitz on 11/19/21.
+
+import Foundation
+
+class GetOfferingsOperation: CacheableNetworkOperation {
+
+    private let offeringsCallbackCache: CallbackCache<OfferingsCallback>
+    private let configuration: AppUserConfiguration
+
+    init(configuration: UserSpecificConfiguration,
+         offeringsCallbackCache: CallbackCache<OfferingsCallback>) {
+        self.configuration = configuration
+        self.offeringsCallbackCache = offeringsCallbackCache
+
+        super.init(configuration: configuration, individualizedCacheKeyPart: configuration.appUserID)
+    }
+
+    override func begin() {
+        self.getOfferings()
+    }
+
+}
+
+private extension GetOfferingsOperation {
+
+    func getOfferings() {
+        guard let appUserID = try? configuration.appUserID.escapedOrError() else {
+            self.offeringsCallbackCache.performOnAllItemsAndRemoveFromCache(withCacheable: self) { callback in
+                callback.completion(nil, ErrorUtils.missingAppUserIDError())
+            }
+            return
+        }
+
+        let path = "/subscribers/\(appUserID)/offerings"
```

Comment:
As others have mentioned I think a nice follow up would be abstracting the paths instead of having them in every operations. But this PR is already such a nice improvement.

Reply:
Totally agree.